drivers/SmartThings/matter-switch/src/init.lua

```diff
@@ -464,51 +470,48 @@ local function initialize_switch(driver, device)
   local profile_name = nil
 
   local component_map = {}
-  local current_component_number = 2
   local component_map_used = false
+  local current_component_number = 1
   local parent_child_device = false
 
-  if #switch_eps == 0 and #button_eps == 0 then
-    return
-  end
-
   -- Since we do not support bindings at the moment, we only want to count clusters
   -- that have been implemented as server. This can be removed when we have
   -- support for bindings.
   local num_switch_server_eps = 0
   local main_endpoint = find_default_endpoint(device)
-  if #switch_eps > 0 then
-    for _, ep in ipairs(switch_eps) do
-      if device:supports_server_cluster(clusters.OnOff.ID, ep) then
-        num_switch_server_eps = num_switch_server_eps + 1
-        local name = string.format("%s %d", device.label, num_switch_server_eps)
-        if ep ~= main_endpoint then -- don't create a child device that maps to the main endpoint
-          local child_profile = assign_child_profile(device, ep)
-          driver:try_create_device(
-            {
-              type = "EDGE_CHILD",
-              label = name,
-              profile = child_profile,
-              parent_device_id = device.id,
-              parent_assigned_child_key = string.format("%d", ep),
-              vendor_provided_label = name
-            }
-          )
-          parent_child_device = true
-        end
+
+  -- If button endpoints are present, use MCD for the main endpoint and all button
+  -- endpoints. Note that if switch endpoints are present, the first switch
+  -- endpoint will be considered the main endpoint. Otherwise, the first button
+  -- endpoint will be considered the main endpoint.
+  if #button_eps == 1 or tbl_contains(STATIC_BUTTON_PROFILE_SUPPORTED, #button_eps) then
```

Comment:
You could just add `1` to the list of supported button profiles in `STATIC_BUTTON_PROFILE_SUPPORTED`

Reply:
I decided to do it this way because `STATIC_BUTTON_PROFILE_SUPPORTED` is used later to determine if a profile change is needed for pure button devices (and doesn't include `1` on purpose). See https://github.com/SmartThingsCommunity/SmartThingsEdgeDrivers/blob/41c5e8529faba5d7c6efce0a6a0981c46c2857bd/drivers/SmartThings/matter-switch/src/init.lua#L563

Reply:
I actually decided to make this change because it simplified logic in other places that use this table.